ArcGISRuntimeSDKQt_QMLSamples/Features/FeatureLayer_Query/FeatureLayer_Query.qml

```diff
@@ -154,13 +126,62 @@ Rectangle {
         }
     }
 
-    // function to form and execute the query
-    function query() {
-        // set the where clause
-        params.whereClause = "STATE_NAME LIKE '" + formatStateNameForQuery(findText.text) + "%'";
+    function selectResults(result, layer) {
```

Comment:
I would opt to leave this method as-is and keep returning `new Promise`. It would be confusing to me to see a standalone method that has `resolve(features)` at the end.

I am not an expert in JS promise, so I don't know the best practices. Also unclear to me if doing it this way causes any issues or not.

Reply:
To be clear what I'm suggesting, if it works, would also remove the `resolve` call. I.E:

```qml

    function selectResults(result, layer) {
                    // clear any previous selection
                    layer.clearSelection();

                    // get the features
                    const features = []
                    while (result.iterator.hasNext) {
                        features.push(result.iterator.next());
                    }

                    // select the features
                    // The ideal way to select features is to call featureLayer.selectFeaturesWithQuery(), which will
                    // automatically select the features based on your query.  This is just a way to show you operations
                    // that you can do with query results. Refer to API doc for more details.
                    layer.selectFeatures(features);
                    return features;
    }
```

And there'd be no `resolve()` call.

Reply:
yup that works, thanks @anmacdonald

Reply:
@ldanzinger, I believe if you throw, it will be consumed by the promise as part of the failure chain.

```qml
    function selectResults(result, layer) {
                    // clear any previous selection
                    layer.clearSelection();

                    // get the features
                    const features = Array.from(result.iterator.features);
                    if (features.length === 0) {
                        const msg = "No state named " + findText.text.toUpperCase() + " exists."
                        throw {message: msg};
                    }

                    // select the features
                    // The ideal way to select features is to call featureLayer.selectFeaturesWithQuery(), which will
                    // automatically select the features based on your query.  This is just a way to show you operations
                    // that you can do with query results. Refer to API doc for more details.
                    layer.selectFeatures(features);

                    return features;
    }
```

Not requesting this change, putting it here as an FYI.